gobblin-core/src/main/java/gobblin/converter/filter/AvroSchemaFieldRemover.java

```diff
@@ -99,6 +99,9 @@ private Schema removeFields(Schema schema, Map<String, Schema> schemaMap) {
   }
 
   private Schema removeFieldsFromRecords(Schema schema, Map<String, Schema> schemaMap) {
+    Schema newRecord = Schema.createRecord(schema.getName(), schema.getDoc(), schema.getNamespace(), schema.isError());
```

Comment:
This is the way Camus does it. The original code failed in one of the tests with infinite recursion. I'm not yet exactly sure why it is the case since that test case was a bit complicated.